central/complianceoperator/v2/report/manager/generator/report_gen_impl.go

```diff
@@ -116,17 +121,19 @@ func (rg *complianceReportGeneratorImpl) ProcessReportRequest(req *report.Reques
 	log.Infof("Sending email for scan config %s", reportName)
 	errC := rg.reportSender.SendEmail(reportGenCtx, reportName, zipData, reportData, req.Notifiers)
 	handler, err := rg.newHandlerFn(func(err error) error {
```

Comment:
I know that the changes of this PR are minimal here, but the entire block starting with`rg.newHandlerFn(` is hard to read and follow. We are passing 2 anonymous functions to another function that is stored in a variable. None of them are named or commented, so to understand what they do, the reader must follow the execution path and try to deduce what happens.

Reply:
I think some small improvements are possible here:
- Let's comment [here](https://github.com/stackrox/stackrox/blob/438ccf3bd9ff9ab92913561c3708626d6e296bf6/central/complianceoperator/v2/report/manager/generator/report_gen_impl.go#L43) what the first and the second function does.
- Let's name the anonymous functions that are passed to `newHandlerFn`

Shall those two anonymous funcs be tested? If yes, then they must not be anonymous.

Reply:
I've de-anonymized the callbacks. I think with the name it's more clear what they do.

> Shall those two anonymous funcs be tested?

The functions are tested in the scope of `TestProcessReportRequest`, we assert that the state of the snapshot is changed depending on the callback execution.